loader/CMakeLists.txt

```diff
@@ -451,6 +451,7 @@ add_library(Vulkan::Loader ALIAS vulkan)
 
 if (APPLE AND BUILD_STATIC_LOADER)
     # When exporting a static library all linked libraries - private or not - need to be exported.
+    install(TARGETS vulkan)
```

Comment:
`BUILD_STATIC_LOADER`
> This allows the loader to be built as a static library on macOS. Not tested, use at your own risk.

My understanding is that BUILD_STATIC_LOADER isn't meant to actually be shipped / installed. It was developed in mind for chromium developers (See BUILD.gn).

Why does Conan want to ship a static version of the vulkan library only for Mac?

Homebrew doesn't do this:
https://github.com/Homebrew/homebrew-core/blob/master/Formula/v/vulkan-loader.rb

So why does Conan need it.

Reply:
conan and homebrew are unrelated (language package manager vs system package manager). conan ships what is possible to ship if user request it. In conan recipe, vulkan-loader is shared by default, but user can ask for static flavor if needed.

Reply:
Yes but this setting isn't really intended for users to ship with. Users really shouldn't use a static version of the vulkan loader. Chromium gets away with it because they always pull in the latest version of the loader. Other projects shouldn't use this.

Reply:
It's basically a feature intended for loader development specifically for the chromium project. Conan really shouldn't be using it.

Reply:
But if Vulkan-Loader developers say that static Vulkan-Loader must not be used, we can remove shared option in its conan recipe, so that shared is always provided and users can't decide.

Reply:
That would make me happy. Static build really only exists at the request of google, and since everything on apple goes in a bundle, I see very little reason to need it over shared.

Reply:
Ok, static will be removed in conan recipe after https://github.com/conan-io/conan-center-index/pull/20899

Reply:
So if you don't want to provide a way to install static lib through CMake, it would be nice to update this message `message(NOTICE "vulkan.pc and VulkanLoaderConfig.cmake not generated for STATIC Apple builds!")`, don't you think?

Reply:
Sounds good to me. I'll make another PR with a better error message based on your feedback.

Reply:
I just want to add my 2 cents and say I really don't understand the special treatment of the static build: only for Apple and then only for the Chromium but let's also mask it out of the GUI and make it _super secret_ and unofficial so no one else gets tempted to use it and it's not _the way_. I use a static loader too for [static-wine32](https://github.com/MIvanchev/static-wine32) and it works great. A lot of other folks have their perfectly good reasoning to use it in their projects and are aware of the implications. The discussions in the issues come up at regular intervals for a reason. After all the loader is nothing but piece of code loading drivers and layers.

Reply:
The Vulkan-Loader is a 'driver front-end', not a regular application library. 

When an application statically links the loader, they are effectively statically linking a graphics driver. Multiple times in the past the way drivers are located on windows has changed, rendering older applications that statically linked unable to find drivers. This is the reason static linking was removed from this project. 

MacOS is special simply because applications *do* ship their driver with the app, as MoltenVK is the driver. As such, shipping the loader in an application is fine since there is only ever one driver to find. 

Just because "it works for you" doesn't mean it works for everyone and will work into the future. The decision to remove static builds of the loader outside of MacOS was deliberate.

Reply:
I understand all of these points however in general I think there is / will be definitely need for an integrated static build. static-wine32 for instance is a good option for outdated distros.

> MacOS is special simply because applications do ship their driver with the app, as MoltenVK is the driver. As such, shipping the loader in an application is fine since there is only ever one driver to find.

Raspberry Pi and many other platforms could profit from this as well. Oftern dynamic linking is more of a nuisance than a good solution.

>Multiple times in the past the way drivers are located on windows has changed, rendering older applications that statically linked unable to find drivers.

I mean, given Microsoft's economic capacity they should be able to plan and agree with OEMs on a mechanism and not change it all the time but, yeah, I understand very well, you don't have the easiest job sitting in the middle =)